bms_current_range-test.cpp

```diff
@@ -0,0 +1,24 @@
+#define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
+
+#include "catch.hpp"
+#include "bms_current_range.h"
+#include "stdlib.h"
+#include "math.h"
+
+/* TEST_CASE("1. Fail Test case to check NULL values")
+{
+    int readingSet[] = {0};
+     Status testStatus;
+    int setLength = sizeof(readingSet) / sizeof(readingSet[0]);
+    testStatus  = checkAndCountCurrentRanges(NULL, setLength);
+    REQUIRE(testStatus == TEST_FAIL);
```

Comment:
This test fails because the test itself is wrong.
Make a test that is correct (according to requirements / statement of a functionality) and have it fail due to incomplete implementation.